pkg/gatewayserver/io/udp/udp.go

```diff
@@ -171,10 +171,9 @@ func (s *srv) handlePackets() {
 
 			if s.firewall != nil {
 				if err := s.firewall.Filter(packet); err != nil {
-					if errors.IsResourceExhausted(err) && ratelimit.Require(s.limitLogs, ratelimit.NewCustomResource(eui.String())) == nil {
-						break
+					if !errors.IsResourceExhausted(err) || ratelimit.Require(s.limitLogs, ratelimit.NewCustomResource(eui.String())) == nil {
```

Comment:
So we never want to log the `ResourceExhausted` error?

Wouldn't the following work for our case?
```
errors.IsResourceExhausted(err) && ratelimit.Require(s.limitLogs, ratelimit.NewCustomResource(eui.String())) != nil
```

Reply:
If the error is not `ResourceExhausted`, we always log (`||` shorts).

If the error is `ResourceExhausted`, the first part of the `if` is false, so we check the rate limitting. If that one is fine, we log.